src/lib/ledger_catchup/normal_catchup.ml

```diff
@@ -493,7 +493,9 @@ let verify_transitions_and_build_breadcrumbs ~logger
              ledger catchup: $error" ;
           Deferred.Or_error.fail
             (Error.tag ~tag:"verifier threw an error" error)
-      | Error `Invalid_proof ->
+      | Error `Invalid_proof
+      | Error `Invalid_block_creator
+      | Error `Invalid_reference_signature ->
```

Comment:
We should give different error strings for each of these cases instead of reporting them all as "invalid proof".